### ISSUE TITLE
Magnifier - Bad focus alignment

### ISSUE THREAD

**Issue by CyrilleB79** [nvaccess/nvda#20196]
When using NVDA Magnifier, the focus is centered. It should be left aligned (or at least configurable) since in list views the left-most part of the focused element is the most interesting one.

#### STR1
* Start Magnifier at high magnification level, e.g. 6x
* Open the add-on store and go to the list
* Browse the add-ons list
#### STR 2
* Open NVDA's settings, e.g. Vision settings
* Tab through Visual Highlight checkboxes
### Actual behavior
STR1: The focus is too wide for the screen and magnified view is centered on its center. Thus, the add-on's name is not visible; only the version the channel and a part of the author field are visible.

STR2: When tabbing, the checkboxes are not on the same vertical line. Some are even off-screen.
### Expected behavior
STR1: The left part of the focus should be visible since in a list view (and in many controls) it's the most important part of the control.

STR2: The X position of the checkboxes should remain the same when tabbing since the eye expects to go to next checkbox which is vertically aligned.

It's worth noting that focus is not centered but left aligned in Windows Magnifier and probably in other magnifiers too.
### NVDA logs, crash dumps and other attachments
N/A
#### NVDA installed/portable/running from source
Source
#### NVDA version
Last beta (source-beta-4b29421 (2026.2.0.0))
#### Windows version
Windows 11 23H2 (AMD64) build 22631.6783
#### Name and version of other software in use when reproducing the issue
N/A
#### Other information about your system
N/A
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors
Previous NVDA Magnifier versions show the same issue.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes
#### Does the issue still occur after you run the System Accessibility Repair Tool in NVDA's tools menu?
Not tested

**Comment by seanbudd** [comment]
We will need to consider right to left languages when doing this

**Comment by Boumtchack** [comment]
I'm doing this, and I was wondering if we should just go for top left or if I could make a condition so it's more intelligent than just this: 
if the object is wider than taller: we center on left and center of y
if the object is taller than wider: we center on top and center of x

so that we are always centered in a "smarter way"

if any both are bigger than the magnified view we fallback to top/left

regarding Sean message, we could later put on option to choose reading direction so it switch left with right

**Comment by CyrilleB79** [comment]
It would be nice to benchmark other magnifiers (e.g. Windows Magnifier and Zoomtext).

Windows Magnifier is already quite comfortable. I do not think it's using focus centering, but I do not exactly what its algo is. On contrary to mouse or text caret, there is no option to specifyfor focus tracking, so there is only one tracking algo for focus.

To test its algo, I suggest to disable all its tracking (mouse, text cursor and narrator) except focus. And test with:
* low zoom level (e.g. 2x-3x)  vs high ones (e.g. 6x-8x)
* size of the focused controls (small / big in both dimensions (veritcal / horizontal))
* manually move move the magnifier view (with `control+alt+arrows`) and then trigger a focus change
* test if the type of focused element changes something, e.g. are menu bars always aligned top?
* if the container of the focus changes something, e.g. group for radio buttons, dialogs, etc.

IIRC, Zoomtext tracking is more complex / configurable. You can read its documentation or even install it and test it without any licence during 40 minutes after you restart your computer.

**Comment by CyrilleB79** [comment]
At first glance, I feel that centering the focused element is not a good algo since it cause the issue reported here in STR2, i.e. when focus jumps down vertically the checkboxes / label start do not remain vertically aligned.

**Comment by Boumtchack** [comment]
I've made some test and I think keeping it simple with top/left is good, we can add options if some edge cases are discovred
